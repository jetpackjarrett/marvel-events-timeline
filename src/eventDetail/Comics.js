import React, { PropTypes } from 'react';
import { getImageUrl } from '../services/marvelApi';

const Comics = ({ comics  }) => {
  return (
    <section>
      <h4>Comics</h4>
      <div className="event-detail-comics">
        {comics ? (
          comics
            .filter((comic) => comic.issueNumber > 0)
            .map(({ id, thumbnail, title, urls }) => {
              return (
                <div key={id} className="event-detail-comic">
                  <a rel="noopener noreferrer" target="_blank" href={urls[0].url}>
                    <img src={getImageUrl(thumbnail, 'portrait_medium')} alt={title} />
                  </a>
                </div>
              );
            })
          ) : (
            <div className="loader" />
          )
        }
      </div>
    </section>
  );
};

Comics.propTypes = {
  comics: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    issueNumber: PropTypes.number,
    thumbnail: PropTypes.shape({
      extension: PropTypes.string.isRequired,
      path: PropTypes.string.isRequired,
    }).isRequired,
    title: PropTypes.string.isRequired,
    urls: PropTypes.array.isRequired,
  })),
};

export default Comics;
