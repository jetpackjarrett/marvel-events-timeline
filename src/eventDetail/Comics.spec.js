/*eslint import/no-extraneous-dependencies: ["error", { "devDependencies": true }] */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Comics from './Comics';

describe('EventDetail Comics', () => {
  it('renders comics thumbnails from the given comics array', () => {
    const component = shallow(
      <Comics
        comics={[
          {
            id: 111,
            issueNumber: 1,
            title: 'Amazing Javascript #1',
            thumbnail: {
              extension: '.jpg',
              path: 'http://mock-image.com/fake',
            },
            urls: [{ url: 'http://mock-url-ajs.com' }],
          },
          {
            id: 222,
            issueNumber: 2,
            title: 'Amazing Javascript #2',
            thumbnail: {
              extension: '.jpg',
              path: 'http://mock-image.com/fake2',
            },
            urls: [{ url: 'http://mock-url-ajs2.com' }],
          },
        ]} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('when no comics prop is supplied, the loader is shown instead', () => {
    const component = shallow(
      <Comics />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
