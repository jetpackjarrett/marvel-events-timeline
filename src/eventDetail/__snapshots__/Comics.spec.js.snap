exports[`EventDetail Comics renders comics thumbnails from the given comics array 1`] = `
<section>
  <h4>
    Comics
  </h4>
  <div
    className="event-detail-comics">
    <div
      className="event-detail-comic">
      <a
        href="http://mock-url-ajs.com"
        rel="noopener noreferrer"
        target="_blank">
        <img
          alt="Amazing Javascript #1"
          src="http://mock-image.com/fake/portrait_medium..jpg" />
      </a>
    </div>
    <div
      className="event-detail-comic">
      <a
        href="http://mock-url-ajs2.com"
        rel="noopener noreferrer"
        target="_blank">
        <img
          alt="Amazing Javascript #2"
          src="http://mock-image.com/fake2/portrait_medium..jpg" />
      </a>
    </div>
  </div>
</section>
`;

exports[`EventDetail Comics when no comics prop is supplied, the loader is shown instead 1`] = `
<section>
  <h4>
    Comics
  </h4>
  <div
    className="event-detail-comics">
    <div
      className="loader" />
  </div>
</section>
`;
